lib/cucumberEventListener.js

```diff
@@ -95,13 +95,22 @@ export class CucumberEventListener extends EventEmitter {
 
         const doc = this.gherkinDocEvents.find(gde => gde.uri === uri).document
         const feature = doc.feature
-        const scenario = doc.feature.children.find(child => child.location.line === sourceLocation.line)
-        const preparedTestCase = this.testCasePreparedEvents.find(tcpe => tcpe.sourceLocation.uri === uri && tcpe.sourceLocation.line === sourceLocation.line)
-        const preparedTestCaseStep = preparedTestCase.steps[testStepStartedEvent.index]
-        const allSteps = feature.children.map(child => child.steps).reduce((list, childSteps) => list.concat(childSteps), [])
-        const step = allSteps.find(childStep => childStep.location.line === preparedTestCaseStep.sourceLocation.line)
+        const scenario = doc.feature.children.find((child) => {
+            if (child.type.indexOf('ScenarioOutline') > -1) {
+                return child.examples[0].tableBody.some(function (tableEntry) {
+                    return tableEntry.location.line === sourceLocation.line
+                })
+            } else {
+                return child.location.line === sourceLocation.line
+            }
+        })
+        let combinedSteps = []
+        feature.children.forEach(function (child) {
```

Comment:
Can you make this an arrow function like this like previous remark?

Reply:
will do tomorrow :)